internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -498,6 +499,7 @@ func (c *enrollCmd) enroll(ctx context.Context, persistentConfig map[string]inte
 		Metadata: fleetapi.Metadata{
 			Local:        metadata,
 			UserProvided: c.options.UserProvidedMetadata,
+			Tags:         c.options.Tags,
 		},
```

Comment:
We should probably run all the tags through `strings.TrimSpace` and validate we aren't sending any empty string?